sdk-base/src/main/java/com/mapbox/maps/plugin/animation/MapboxAnimatorThread.kt

```diff
@@ -0,0 +1,184 @@
+   * queue. This is to reduce thread context switches while running animations.
+   */
+  fun post(runnable: Runnable): Boolean {
+    return (backgroundAnimatorThread ?: mainAnimatorThread).post(runnable).also { posted ->
+      if (!posted) {
+        logW(
+          TAG,
+          "post did not happen, ensure the app is in foreground. background  animator " +
+            "thread: ${backgroundAnimatorThread != null}"
+        )
+      }
+    }
+  }
+
+  /**
+   * Mapbox gl-native libraries require the main thread. Use this function whenever an animation
+   * task should be posted on the main thread, even when background threads are enabled.
+   *
+   * Run immediate if my looper is the designated looper, otherwise post the work to a handler
+   * queue. This is to reduce thread context switches while running animations.
+   */
+  fun postMain(runnable: Runnable): Boolean {
+    return mainAnimatorThread.post(runnable).also { posted ->
+      if (!posted) {
+        logW(TAG, "postMain did not happen, ensure the app is in foreground")
+      }
+    }
+  }
+
+  /**
+   * The beginning of the animator thread lifecycle. This will create resources for running
+   * animator threads.
+   */
+  fun onDetached() {
+    mainAnimatorThread.onDetached()
+    backgroundAnimatorThread?.onDetached()
+    backgroundAnimatorThread = null
+  }
+
+  private companion object {
+    private const val TAG = "MapboxAnimatorThread"
+  }
+}
+
+/**
+ * The [PackageManager] has issues being mocked. This class makes it so usages of the package
+ * meta data can be tested without mocking the application info metadata.
+ *
+ * https://github.com/mockk/mockk/issues/182
+ */
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+internal class AndroidManifestMetaData {
+
+  private fun metaData(): Bundle {
+    val context = MapboxSDKCommon.getContext()
+    val packageManager: PackageManager = context.packageManager
+    return packageManager.getApplicationInfo(
+      context.packageName,
+      PackageManager.GET_META_DATA
+    ).metaData
+  }
+
+  /**
+   * Perform manifest metadata lookup for boolean value of
+   * com.mapbox.maps.BackgroundAnimatorThread. Returns false by default.
+   */
+  fun backgroundAnimatorThreadEnabled(): Boolean {
+    return metaData().getBoolean(BACKGROUND_ANIMATOR_THREAD_KEY, false)
+  }
+
+  private companion object {
+    private const val BACKGROUND_ANIMATOR_THREAD_KEY = "com.mapbox.maps.BackgroundAnimatorThread"
+  }
+}
+
+private interface AnimatorThread {
+  @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+  val handler: Handler?
+
+  fun onAttached(): AnimatorThread
+  fun onDetached()
+
+  fun post(runnable: Runnable): Boolean {
+    return if (Looper.myLooper() == handler?.looper) {
+      runnable.run()
+      true
+    } else {
+      handler?.post(runnable) ?: false
+    }
+  }
+}
+
+private class BackgroundAnimatorThread : AnimatorThread {
```

Comment:
I'd split those classes to separate files for better readability

Reply:
Yeah I was playing around with this for some time yesterday, trying to make it more readable.

What exactly is the suggestion? The classes are broken up a bit (`MapboxAnimatorThread` is using 2 other classes, at first everything was in the `MapboxAnimatorThread`)

Adding the `BackgroundAnimatorThread` made `MapboxAnimatorThread` simple
``` kotlin
  fun post(runnable: Runnable): Boolean {
    return (backgroundAnimatorThread ?: mainAnimatorThread).post(runnable)
  }
```

Reply:
Yeah needed to move packages, moved it all to the `:plugin-animations:` module